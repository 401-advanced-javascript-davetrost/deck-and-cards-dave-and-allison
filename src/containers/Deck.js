import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class Deck extends Component {

  render() {
    return (
      <section>
        <header>
          <h2>{this.props.title}</h2>
        </header>
        {this.props.children}
      </section>
    );
  }

  static propTypes = {
    title: PropTypes.string,
    children: PropTypes.node.isRequired
  };
}

