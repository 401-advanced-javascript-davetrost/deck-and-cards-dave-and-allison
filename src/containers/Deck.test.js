import React from 'react';
import { shallow } from 'enzyme';
import Deck from './Deck';

describe('Deck component', () => {
  it('renders', () => {
    const wrapper = shallow(
      <Deck title="Test Title">
        <p>card 1</p>
        <p>card 2</p>
      </Deck>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders without a title', () => {
    const wrapper = shallow(
      <Deck>
        <p>card 1</p>
        <p>card 2</p>
      </Deck>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
