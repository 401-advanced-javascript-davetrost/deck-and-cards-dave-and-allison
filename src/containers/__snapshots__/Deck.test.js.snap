// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deck component renders 1`] = `
<section>
  <header>
    <h2>
      Test Title
    </h2>
  </header>
  <p>
    card 1
  </p>
  <p>
    card 2
  </p>
</section>
`;
